lib/elixir/src/elixir_expand.erl

```diff
@@ -1134,6 +1134,8 @@ format_error({invalid_call, Call}) ->
   io_lib:format("invalid call ~ts", ['Elixir.Macro':to_string(Call)]);
 format_error({invalid_quoted_expr, Expr}) ->
   io_lib:format("invalid quoted expression: ~ts", ['Elixir.Kernel':inspect(Expr, [])]);
+format_error({invalid_local_invocation, guard, {Name, _, Args} = Call}) ->
+  io_lib:format("cannot find local ~ts/~B", [Name, length(Args)]);
 format_error({invalid_local_invocation, Context, {Name, _, Args} = Call}) ->
   io_lib:format("cannot invoke local ~ts/~B inside ~ts, called as: ~ts",
```

Comment:
I think it would be better to change this message, instead of special casing the guard one, because can have custom macros in both match and guard. Also note that the old message is still valid, so we need to make sure to keep it to. I propose something like:

    cannot find or cannot invoke local  ~ts/~B inside ~ts.  Only macros can be invoked in a ~ts and they must be defined before their invocation. Called as: ~ts

What do you think?